src/realm/util/file.cpp

```diff
@@ -1851,7 +1868,33 @@ File::SizeType File::get_free_space(const std::string& path)
 #endif
 }
 
-#ifndef _WIN32
+#if REALM_HAVE_STD_FILESYSTEM
+
+DirScanner::DirScanner(const std::string& path, bool allow_missing)
+{
+    try {
+        m_iterator = std::filesystem::directory_iterator(path);
+    }
+    catch (const std::filesystem::filesystem_error& e) {
+        if (e.code() != std::errc::no_such_file_or_directory || !allow_missing)
+            throw;
+    }
+}
+
+DirScanner::~DirScanner() = default;
+
+bool DirScanner::next(std::string& name)
+{
+    const std::filesystem::directory_iterator end;
+    if (m_iterator != end) {
+        name = m_iterator->path().filename().string();
+        m_iterator++;
+        return true;
+    }
+    return false;
```

Comment:
```suggestion
    if (m_iterator == end)
        return false;
    name = m_iterator->path().filename().string();
    m_iterator++;
    return true;
```
as a general rule, putting the early-exit case in the if block makes code easier to read since it requires maintaining less info in your mental stack.